File: Practice/print.c
#include <stdio.h>
#include <string.h>

void reverseStr(char str[], int index)
{
    if (index == 0)
    {
        printf("%c ", str[index]);
        return;
    }

    printf("%c ", str[index]);
    reverseStr(str, index - 1);
}

int main()
{
    char str[] = "ABCD";

    reverseStr(str, strlen(str) - 1);

    return 0;
}

File: Practice/print.java
public class print {
    public static void reverseStr(String str, int index) {
        if (index == 0) {
            System.out.print(str.charAt(index) + " ");
            return;
        }

        System.out.print(str.charAt(index) + " ");
        reverseStr(str, index - 1);
    }

    public static void main(String[] args) {
        String str = "ABCD";
        reverseStr(str, str.length() - 1);
    }
}